src/main/java/io/cdap/plugin/servicenow/source/ServiceNowMultiInputFormat.java

```diff
@@ -0,0 +1,161 @@
+  private static final Logger LOG = LoggerFactory.getLogger(ServiceNowInputFormat.class);
+
+  /**
+   * Updates the jobConfig with the ServiceNow table information, which will then be read in getSplit() function.
+   *
+   * @param jobConfig the job configuration
+   * @param conf the database conf
+   * @return Collection of ServiceNowTableInfo containing table and schema.
+   */
+  public static Set<ServiceNowTableInfo> setInput(Configuration jobConfig,
+                                                   ServiceNowMultiSourceConfig conf) {
+    ServiceNowJobConfiguration jobConf = new ServiceNowJobConfiguration(jobConfig);
+    jobConf.setMultiSourcePluginConfiguration(conf);
+
+    // Depending on conf value fetch the list of fields for each table and create schema object
+    // return the schema object for each table as ServiceNowTableInfo
+    Set<ServiceNowTableInfo> tableInfos = fetchTablesInfo(conf);
+
+    jobConf.setTableInfos(tableInfos.stream().collect(Collectors.toList()));
+
+    return tableInfos;
+  }
+
+  public static Set<ServiceNowTableInfo> fetchTablesInfo(ServiceNowMultiSourceConfig conf) {
+
+    Set<ServiceNowTableInfo> tablesInfos = new LinkedHashSet<>();
+
+    Set<String> tableNames = getList(conf.getTableNames());
+    for (String tableName : tableNames) {
+      ServiceNowTableInfo tableInfo = getTableMetaData(tableName, conf);
+      if (tableInfo == null) {
+        continue;
+      }
+      tablesInfos.add(tableInfo);
+    }
+
+    return tablesInfos;
+  }
+
+  private static ServiceNowTableInfo getTableMetaData(String tableName, ServiceNowMultiSourceConfig conf) {
+    // Call API to fetch first record from the table
+    ServiceNowTableAPIClientImpl restApi = new ServiceNowTableAPIClientImpl(conf);
+
+    ServiceNowTableDataResponse response = restApi.fetchTableSchema(tableName, conf.getStartDate(), conf.getEndDate(),
+      true);
+    if (response == null) {
+      return null;
+    }
+
+    List<ServiceNowColumn> columns = response.getColumns();
+    if (columns == null || columns.isEmpty()) {
+      return null;
+    }
+
+    SchemaBuilder schemaBuilder = new SchemaBuilder();
+    Schema schema = schemaBuilder.constructSchema(tableName, columns);
+    LOG.debug("table {}, rows = {}", tableName, response.getTotalRecordCount());
+    return new ServiceNowTableInfo(tableName, schema, response.getTotalRecordCount());
+  }
+
+  @Override
+  public List<InputSplit> getSplits(JobContext jobContext) throws IOException, InterruptedException {
+    ServiceNowJobConfiguration jobConfig = new ServiceNowJobConfiguration(jobContext.getConfiguration());
+
+    List<ServiceNowTableInfo> tableInfos = jobConfig.getTableInfos();
+    List<InputSplit> resultSplits = new ArrayList<>();
+
+    for (ServiceNowTableInfo tableInfo : tableInfos) {
+      String tableName = tableInfo.getTableName();
+      int totalRecords = tableInfo.getRecordCount();
+      if (totalRecords <= ServiceNowConstants.PAGE_SIZE) {
```

Comment:
you can remove this if condition as the single split case is already handled below

Reply:
Removed. Thanks! 7661e08